include/vast/CodeGen/CodeGenAttrVisitor.hpp

```diff
@@ -65,5 +65,26 @@ namespace vast::cg {
         mlir_attr VisitWarnUnusedResultAttr(const clang::WarnUnusedResultAttr *attr) {
             return make< hl::WarnUnusedResultAttr >();
         }
+
+        mlir_attr VisitRestrictAttr(const clang::RestrictAttr *attr) {
+            return make< hl::MallocAttr >();
+        }
+
+        mlir_attr VisitNoThrowAttr(const clang::NoThrowAttr *attr) {
+            return make< hl::NoThrowAttr >();
+        }
+
+        mlir_attr VisitBuiltinAttr(const clang::BuiltinAttr *attr) {
+            return make< hl::BuiltinAttr >(attr->getID());
+        }
+
+        mlir_attr VisitAllocSizeAttr(const clang::AllocSizeAttr *attr) {
+            auto size = attr->getElemSizeParam();
+            auto num = attr->getNumElemsParam();
```

Comment:
Use `num = attr->getNumElemsParam().isValid() ? ...`
and only a single `hl::AllocSizeAttr` constructor.

Maybe, i deally monadic builder with binds.